ui/lib/kmip/addon/templates/components/edit-form-kmip-role.hbs

```diff
@@ -9,48 +8,86 @@
         @model={{model}}
       />
     {{/if}}
-    <h3 class="title is-5">
-      Allowed Operations
-    </h3>
-    {{#each (array
-      (hash label="Allow all" value="operationAll")
-      (hash label="Allow none" value="operationNone")
-      (hash label="Let me choose" value="choose")
-      ) as |displayType|}}
-        <RadioButton
-          @value={{displayType.value}}
-          @groupValue={{this.display}}
-          @changed={{queue
-            (action (mut this.display))
-            (action "updateModel")
-          }}
-          @name="role-display"
-          @radioId={{displayType.value}}
-          @classNames="vlt-radio is-block"
-        >
-          <label for={{displayType.value}} />
-          {{displayType.label}}
-        </RadioButton>
-    {{/each}}
-    {{#if (eq this.display "choose")}}
-      <div class="box is-sideless is-shadowless is-marginless">
-        {{#each this.model.operationFormFields as |attr|}}
-          <FormField
-            data-test-field
-            @attr={{attr}}
-            @model={{model}}
-          />
-        {{/each}}
+    <div class="control is-flex box is-shadowless is-fullwidth is-marginless">
+      <input
+        data-test-input="operationNone"
+        id="operationNone"
+        type="checkbox"
+        class="switch is-rounded is-success is-small"
+        checked={{not this.model.operationNone}}
+        onchange={{action "switchUpdated" value="target.checked"}}
+      />
+      <label for="operationNone">
+        Allow this role to perform KMIP operations
+      </label>
+    </div>
+    {{#if (not this.model.operationNone) }}
+      <Toolbar>
+        <h3 class="title is-6">
+          Allowed Operations
+        </h3>
+      </Toolbar>
+      <div class="box">
+        <FormField
+          @attr={{hash name="operationAll" type="boolean" options=(hash label="Allow this role to perform all operations")}}
+          @model={{this.model}}
+        />
+        <hr />
+        <div class="is-flex">
+          <div class="kmip-role-allowed-operations">
+            {{#each-in this.model.operationFormFields.firstObject as |groupName fieldsInGroup|}}
+              <h4 class="title is-7">{{groupName}}</h4>
+              {{#each fieldsInGroup as |attr|}}
+                <FormField
+                  data-test-field
+                  @disabled={{or this.model.operationNone this.model.operationAll}}
+                  @attr={{attr}}
+                  @model={{model}}
+                  @showHelpText={{false}}
+                />
+              {{/each}}
+            {{/each-in}}
+          </div>
+          <div class="kmip-role-allowed-operations">
+            {{#each (drop 1 (or this.model.operationFormFields (array))) as |group|}}
```

Comment:
what does the `drop` template helper do? I can't find it anywhere.

Reply:
It's from ember-compostable helpers - https://github.com/DockYard/ember-composable-helpers#drop - it drops N items from the front of an array - here we're doing it because the first item is rendered separately (because of the containing div).